From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: pkgdemon <pkgdemonteam@gmail.com>
Date: Sat, 24 Oct 2026 01:44:48 -0500
Subject: [PATCH] Only check the variadic arguments in ffi_prep_cif_var

Upstream libffi commit de95947a ("Fix check for invalid varargs
arguments", #707, libffi 3.4.3).

3.4.0's new check that rejects float and sub-int arguments in the
variadic part of a call started at argument 1, not at nfixedargs, so
ffi_prep_cif_var returned FFI_BAD_ARGTYPE for a legal call with a fixed
float or char argument after the first.
---
 src/prep_cif.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/prep_cif.c b/src/prep_cif.c
index 3f07583..6d38097 100644
--- a/src/prep_cif.c
+++ b/src/prep_cif.c
@@ -1,6 +1,7 @@
 /* -----------------------------------------------------------------------
    prep_cif.c - Copyright (c) 2011, 2012, 2021  Anthony Green
                 Copyright (c) 1996, 1998, 2007  Red Hat, Inc.
+                Copyright (c) 2022 Oracle and/or its affiliates.
 
    Permission is hereby granted, free of charge, to any person obtaining
    a copy of this software and associated documentation files (the
@@ -240,7 +241,7 @@ ffi_status ffi_prep_cif_var(ffi_cif *cif,
   if (rc != FFI_OK)
     return rc;
 
-  for (i = 1; i < ntotalargs; i++)
+  for (i = nfixedargs; i < ntotalargs; i++)
     {
       ffi_type *arg_type = atypes[i];
       if (arg_type == &ffi_type_float
